### PR TITLE
[Feature][Master] Failover workflow instance base on priority

### LINKED ISSUE #10750: [Feature][Master] Failover workflow  instance in priority
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

When failing over a workflow instance, it should be based on the priority of the instance

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhuxt2015**
## Purpose of the pull request
fix #10750

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

File: dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessInstanceMapper.xml [hunk 1]
```diff
@@ -44,7 +44,7 @@
         <foreach collection="states" item="i" open="(" close=")" separator=",">
             #{i}
         </foreach>
-        order by id asc
+        order by process_instance_priority, id asc
```
Comment (SbloodyS) [inline dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessInstanceMapper.xml]: I'm not quite sure if this is nessnary to do this.

I think we can provide users with a sort item, which is similar to excel and gives users free choice.

WDYT? @zhuxt2015 @zhongjiajie @caishunfeng

Comment (zhuxt2015) [reply dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessInstanceMapper.xml]: I think sort by process_instance_priority is most suitable and most commonly used . why do you  want to add sort item, Can you give some cases?

Comment (ruanwenjun) [reply dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessInstanceMapper.xml]: I think we don't need to change this part, we need to set the processInstance's priority in to failover command is enough. if you query by priority here, you still need to set the priority of command. The problem is that we didn't set the priority in to failover command.

Comment (SbloodyS) [reply dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessInstanceMapper.xml]: Sort item like this.

**Review by ruanwenjun — APPROVED** [review]
LGTM

**Review by SbloodyS — APPROVED** [review]
+1

**Pull request merged**